cache/network/download.go

```diff
@@ -38,48 +38,33 @@ func Download(ctx context.Context, params DownloadParams, logger log.Logger) (ma
 		return "", fmt.Errorf("cache key list is empty")
 	}
 
-	retryableHTTPClient := prepareRetryableHTTPClient(logger)
+	retryableHTTPClient := retryhttp.NewClient(logger)
+	retryableHTTPClient.CheckRetry = customRetryFunction
```

Comment:
This can be deleted because if I see it correctly the default value of the `CheckRetry` property is the `DefaultRetryPolicy` function:
https://github.com/hashicorp/go-retryablehttp/blob/4165cf8897205a879a06b20d1ed0a2a76fbb6a17/client.go#L436C30-L436C30

And our `customRetryFunction` just wraps the default value.

Reply:
Correct, the only reason I re-added `customRetryFunction` is to show that the unit test still passes. But it's true that I could test with `DefaultRetryPolicy` there directly.

The reason why I'd keep these here is because I plan to add some debug logging to the retry func :)